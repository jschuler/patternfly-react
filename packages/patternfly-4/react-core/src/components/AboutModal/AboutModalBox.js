import React from 'react';
import { css } from '@patternfly/react-styles';
import PropTypes from 'prop-types';
import styles from '@patternfly/patternfly-next/components/AboutModalBox/styles.css';

const propTypes = {
  /** content rendered inside the AboutModelBox. */
  children: PropTypes.node.isRequired,
  /** additional classes added to the AboutModalBox */
  className: PropTypes.string,
  /** id to use for Modal Box aria-labelledby attribute */
  ariaLablledbyId: PropTypes.string.isRequired,
  /** id to use for Modal Box aria-describedby attribute */
  ariaDescribedById: PropTypes.string.isRequired
};

const defaultProps = {
  className: ''
};

const AboutModalBox = ({ children, className, ariaLablledbyId, ariaDescribedById, ...props }) => (
  <div
    {...props}
    role="dialog"
    aria-labelledby={ariaLablledbyId}
    aria-describedby={ariaDescribedById}
    aria-modal="true"
    className={css(styles.aboutModalBox, className)}
  >
    {children}
  </div>
);
AboutModalBox.propTypes = propTypes;
AboutModalBox.defaultProps = defaultProps;

export default AboutModalBox;
