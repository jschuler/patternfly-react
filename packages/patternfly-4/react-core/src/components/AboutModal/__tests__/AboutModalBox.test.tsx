import * as React from 'react';
import { shallow } from 'enzyme';
import { AboutModalBox } from '../AboutModalBox';
// any missing imports can usually be resolved by adding them here
import {  } from '..';

it('AboutModalBox should match snapshot (auto-generated)', () => {
  const view = shallow(
    <AboutModalBox
      children={<div>ReactNode</div>}
			className={"''"}
    />);
  expect(view).toMatchSnapshot();
});

test('AboutModalBox Test', () => {
  const view = shallow(
    <AboutModalBox aria-labelledby="id" aria-describedby="id2">
      This is a AboutModalBox
    </AboutModalBox>
  );
  expect(view).toMatchSnapshot();
});