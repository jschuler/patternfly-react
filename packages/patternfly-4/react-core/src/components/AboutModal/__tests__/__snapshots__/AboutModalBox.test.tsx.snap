// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutModalBox Test 1`] = `
<div
  aria-describedby="id2"
  aria-labelledby="id"
  aria-modal="true"
  className="pf-c-about-modal-box"
  role="dialog"
>
  This is a AboutModalBox
</div>
`;

exports[`AboutModalBox should match snapshot (auto-generated) 1`] = `
<div
  aria-modal="true"
  className="pf-c-about-modal-box ''"
  role="dialog"
>
  <div>
    ReactNode
  </div>
</div>
`;
